client/src/main.rs

```diff
@@ -566,10 +515,46 @@ fn fetch(
     } else {
         log::info!("{}", "peers are already up to date.".green());
     }
+
     store.set_cidrs(cidrs);
-    store.update_peers(peers)?;
+    store.update_peers(&peers)?;
     store.write().with_str(interface.to_string())?;
 
+    let candidates = wg::get_local_addrs()?
+        .into_iter()
+        .map(|addr| SocketAddr::from((addr, device.listen_port.unwrap_or(51820))).into())
+        .collect::<Vec<Endpoint>>();
+    log::info!(
+        "reporting {} network interface addresses as ICE candidates...",
+        candidates.len()
+    );
+    log::debug!("candidates: {:?}", candidates);
+    match Api::new(&config.server).http_form::<_, ()>("PUT", "/user/candidates", &candidates) {
+        Err(ureq::Error::Status(404, _)) => {
+            log::warn!("Server doesn't support ICE candidate reporting.")
+        },
+        Err(e) => return Err(e.into()),
+        _ => {},
+    }
+
+    log::debug!("viable ICE candidates: {:?}", candidates);
+
+    let mut nat_traverse = NatTraverse::new(interface, network.backend, &modifications);
+    log::info!(
+        "Attempting to establish connection with {} remaining unconnected peers...",
+        nat_traverse.remaining()
+    );
+    loop {
+        nat_traverse.step()?;
+        if nat_traverse.is_finished() {
+            break;
+        }
+        log::info!(
+            "{} unconnected peers remaining...",
+            nat_traverse.remaining()
+        );
+    }
```

Comment:
Hmm, there might be some situations where that would lead to an eventual connection if a peer is genuinely offline rather than behind a strict NAT.

I'll implement it just since it feels cleaner that way either way.